learning_resources/models.py

```diff
@@ -356,3 +356,36 @@ class ContentFile(TimestampedModel):
     class Meta:
         unique_together = (("key", "run"),)
         verbose_name = "contentfile"
+
+
+class UserList(TimestampedModel):
+    """
+    Similar in concept to a LearningPath: a list of learning resources.
+    However, UserLists are not considered LearningResources because they
+    should only be accessible to the user who created them.
+    """
+
+    author = models.ForeignKey(
+        User, on_delete=models.deletion.CASCADE, related_name="user_lists"
+    )
+    title = models.CharField(max_length=256)
+    description = models.TextField(null=True, blank=True)  # noqa: DJ001
+    topics = models.ManyToManyField(LearningResourceTopic)
+    resources = models.ManyToManyField(
+        LearningResource, through="UserListRelationship", symmetrical=False, blank=True
+    )
+
+
+class UserListRelationship(TimestampedModel):
+    """
+    UserListRelationship model tracks the resources belonging to a UserList
+    and their relative positions in the list.
+    """
+
+    parent = models.ForeignKey(
+        UserList, on_delete=models.deletion.CASCADE, related_name="children"
+    )
+    child = models.ForeignKey(
+        LearningResource, related_name="user_lists", on_delete=models.deletion.CASCADE
+    )
+    position = models.PositiveIntegerField(default=0)
```

Comment:
When I POST a resource a newly-created, empty userlist, it gets position 1 (not 0). Weird.

Reply:
For both learningpaths and userlists, the following code is used, which will make the 1st addition have a position of 1:

```python
        position = (
            items.aggregate(Max("position"))["position__max"] or items.count()
        ) + 1
```